Lib/timeit.py

```diff
@@ -315,65 +370,13 @@ def main(args=None, *, _wrap_timer=None):
         timer = _wrap_timer(timer)
 
     t = Timer(stmt, setup, timer)
-    if number == 0:
-        # determine number so that 0.2 <= total time < 2.0
-        callback = None
-        if verbose:
-            def callback(number, time_taken):
-                msg = "{num} loop{s} -> {secs:.{prec}g} secs"
-                plural = (number != 1)
-                print(msg.format(num=number, s='s' if plural else '',
-                                 secs=time_taken, prec=precision))
-        try:
-            number, _ = t.autorange(callback)
-        except:
-            t.print_exc()
-            return 1
-
-        if verbose:
-            print()
 
     try:
-        raw_timings = t.repeat(repeat, number)
-    except:
+        _run(t, repeat=repeat, number=number, time_unit=time_unit, verbose=bool(verbose), precision=verbose + 3)
```

Comment:
The try-except is now around all code, earlier it was just around the `raw_timings = t.repeat(repeat, number)`. I have no strong preference, but to make probability of accepting the PR higher maybe keep the original behaviour from the CLI exactly the same.